File: code_challenge_solution/javascript/85 Haiku generator.js
// Array of possible lines for each haiku structure
const line1 = ["An old silent pond", "Over the wintry", "I write, erase, rewrite", "Twilight moonlight"];
const line2 = ["A frog jumps into the pond -", "moon tonight -", "rearrange the words", "A path of footprints"];
const line3 = ["Splash! Silence again.", "a flower fails.", "and try to find meaning.", "on the windless sea."];

// Function to randomly select a line from each structure
function generateHaiku() {
  const randomLine1 = line1[Math.floor(Math.random() * line1.length)];
  const randomLine2 = line2[Math.floor(Math.random() * line2.length)];
  const randomLine3 = line3[Math.floor(Math.random() * line3.length)];

  return `${randomLine1}\n${randomLine2}\n${randomLine3}`;
}

// Generate and print a random haiku
console.log(generateHaiku());

File: code_challenge_solution/python/85 Haiku generator.py
import random

def generate_haiku():
    # List of words for each line of the haiku
    line1 = ['Autumn', 'Winter', 'Spring', 'Summer']
    line2 = ['leaves', 'snowflakes', 'blossoms', 'sunlight']
    line3 = ['falling', 'melting', 'blooming', 'shining']

    # Randomly select a word for each line
    word1 = random.choice(line1)
    word2 = random.choice(line2)
    word3 = random.choice(line3)

    # Construct the haiku poem
    haiku = f"{word1} {word2} {word3}"

    return haiku

# Generate and print a haiku poem
haiku = generate_haiku()
print(haiku)